### DISCUSSION TITLE
load & save file via api

### DISCUSSION THREAD

**Discussion by troy351** [eclipse-theia/theia#9306] [Q&A]
### Description
 
I am trying to save code file into database rather than file on disk. i.e. save and load files via api.
To make it simple, There is a json which describe files, any ideas or suggestions about how to integrate with theia?
Or how to hijack all file manipulation?
```json
{
  "file_list": [
    {
      "path": "package.json",
      "content": "{}"
    },
    {
      "path": "src/index.js",
      "content": "export default const demo = 1;"
    }
  ]
}
```
### Reproduction Steps

**OS and Theia version:**

**Diagnostics:**

**Comment by akosyakov** [comment]
Theia editors don't work directly against files, but use `Resource` which are identified by URIs. You need to invent a new URI (via a special scheme for instance) and implement `ResourceProvider`.
